omod/src/main/java/org/openmrs/module/providermanagement/rest/ProviderResource.java

```diff
@@ -0,0 +1,160 @@
+package org.openmrs.module.providermanagement.rest;
+
+import org.openmrs.ProviderAttribute;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.providermanagement.Provider;
+import org.openmrs.module.providermanagement.rest.controller.ProviderManagementRestController;
+import org.openmrs.module.webservices.rest.web.RequestContext;
+import org.openmrs.module.webservices.rest.web.RestConstants;
+import org.openmrs.module.webservices.rest.web.annotation.PropertyGetter;
+import org.openmrs.module.webservices.rest.web.annotation.PropertySetter;
+import org.openmrs.module.webservices.rest.web.annotation.Resource;
+import org.openmrs.module.webservices.rest.web.representation.DefaultRepresentation;
+import org.openmrs.module.webservices.rest.web.representation.FullRepresentation;
+import org.openmrs.module.webservices.rest.web.representation.Representation;
+import org.openmrs.module.webservices.rest.web.resource.impl.DelegatingResourceDescription;
+import org.openmrs.module.webservices.rest.web.resource.impl.MetadataDelegatingCrudResource;
+import org.openmrs.module.webservices.rest.web.response.ResponseException;
+import java.util.Set;
+
+@Resource(name = RestConstants.VERSION_1 + ProviderManagementRestController.PROVIDER_MANAGEMENT_REST_NAMESPACE + "/provider", supportedClass = Provider.class, supportedOpenmrsVersions = {
+        "1.9.* - 9.*" })
+public class ProviderResource extends MetadataDelegatingCrudResource<Provider> {
```

Comment:
This whole class is basically a copy from the main Provider resource RET web services, with the additional of provideRole to reps, and using the Provider class from the PM module rather than the core one.

We *almost* could get away with having this class just override the ProviderResource1_9 and add in the providerRole, but I was running into some casting issues.

Kind of annoying that this now can (and should) be kept in-sync with the ProviderResource1_9, but, at the end of the day, we will likely not be changing the core Provider in the near future--and if we do, we should take the opportunity to completely migrate the provider role stuff to core at that time.